Tests are failing to build with MSVC x86 Release

```
Building from D:\a\1\s\build\msvc32relwithdebinfo
...
[12/62] Building CXX object tests\app\CMakeFiles\witest.app.dir\__\wiTest.cpp.obj
FAILED: tests/app/CMakeFiles/witest.app.dir/__/wiTest.cpp.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1424~1.283\bin\Hostx86\x86\cl.exe  /nologo /TP -DCATCH_CONFIG_FAST_COMPILE -DWIL_FAST_BUILD -DWINAPI_FAMILY=WINAPI_FAMILY_PC_APP -I..\..\include -I..\..\tests\.\workarounds\wrl /DWIN32 /D_WINDOWS /W4 /GR- /EHsc /WX /permissive- /wd4324 /bigobj /d2FH4- /MD /Zi /O2 /Ob1 /DNDEBUG /WX /permissive- /wd4324 /bigobj /d2FH4- /showIncludes /Fotests\app\CMakeFiles\witest.app.dir\__\wiTest.cpp.obj /Fdtests\app\CMakeFiles\witest.app.dir\ /FS -c ..\..\tests\wiTest.cpp
D:\a\1\s\include\wil\win32_helpers.h(538) : error C2220: the following warning is treated as an error
D:\a\1\s\include\wil\win32_helpers.h(538) : warning C4702: unreachable code
```

--- a/tests/wiTest.cpp
+++ b/tests/wiTest.cpp
@@ -2494,7 +2494,8 @@ TEST_CASE("WindowsInternalTests::InitOnceNonTests")
     init = {};
 
     // A thrown exception leaves the object un-initialized
-    REQUIRE_THROWS_AS(winner = wil::init_once(init, [&] { called = true; throw wil::ResultException(E_FAIL); }), wil::ResultException);
+    static volatile bool always_true = true; // So that the compiler can't determine that we unconditionally throw below (warning C4702)
+    REQUIRE_THROWS_AS(winner = wil::init_once(init, [&] { called = true; THROW_HR_IF(E_FAIL, always_true); }), wil::ResultException);
     REQUIRE_FALSE(wil::init_once_initialized(init));
     REQUIRE(called);
     REQUIRE_FALSE(winner);

